var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -99,6 +99,9 @@ class PyDask(PythonPackage):
     depends_on("py-partd@0.3.10:", type=("build", "run"), when="@:2021.3.0 +dataframe")
     # The dependency on py-fsspec is non-optional starting version 2021.3.1
     depends_on("py-fsspec@0.6.0:", type=("build", "run"), when="@:2021.3.0 +dataframe")
+    # Starting with version 2024.3.0, dataframe requires a separate package py-dask-expr
```

Comment:
According to the latest `requires.txt` (below), doesn't this mean `+array` is required when `+dataframe`?

```
[dataframe]
dask[array]
pandas>=2.0
dask-expr<1.2,>=1.1
```

Reply:
Indeed. I am going back in history to find out when this was dependency was added, and then I'll add a conflict to catch this. It's unfortunately not the source of the problem I mentioned in item 2 above, but I found something else that I am exploring now.